### PR TITLE
Circuits the updateding

### PR DESCRIPTION
**Pull request by mystery3525**
## About The Pull Request

(clone of https://github.com/BeeStation/BeeStation-Hornet/pull/8382 without the git screwup)

Brings our circuits more up-to-date
Mostly parts stolen from https://github.com/BeeStation/BeeStation-Hornet/pull/5197

- https://github.com/tgstation/tgstation/pull/59345
- https://github.com/tgstation/tgstation/pull/59348
- https://github.com/tgstation/tgstation/pull/59459
- https://github.com/tgstation/tgstation/pull/59584
- https://github.com/tgstation/tgstation/pull/59635
- https://github.com/tgstation/tgstation/pull/59724
- https://github.com/tgstation/tgstation/pull/59728
- https://github.com/tgstation/tgstation/pull/59866
- https://github.com/tgstation/tgstation/pull/59958
- https://github.com/tgstation/tgstation/pull/59993
- https://github.com/tgstation/tgstation/pull/60087
- https://github.com/tgstation/tgstation/pull/60222
- https://github.com/tgstation/tgstation/pull/60227
- https://github.com/tgstation/tgstation/pull/60240 (partial)
- https://github.com/tgstation/tgstation/pull/60263
- https://github.com/tgstation/tgstation/pull/60338
- https://github.com/tgstation/tgstation/pull/60347
- https://github.com/tgstation/tgstation/pull/60462
- https://github.com/tgstation/tgstation/pull/60494
- https://github.com/tgstation/tgstation/pull/60504
- https://github.com/tgstation/tgstation/pull/60514
- https://github.com/tgstation/tgstation/pull/60557
- https://github.com/tgstation/tgstation/pull/60558
- https://github.com/tgstation/tgstation/pull/60571
- https://github.com/tgstation/tgstation/pull/60590
- https://github.com/tgstation/tgstation/pull/60661
- https://github.com/tgstation/tgstation/pull/60670
- https://github.com/tgstation/tgstation/pull/60787
- https://github.com/tgstation/tgstation/pull/60917
- https://github.com/tgstation/tgstation/pull/61143
- https://github.com/tgstation/tgstation/pull/62302

(retroactive)

- https://github.com/tgstation/tgstation/pull/61554
- https://github.com/tgstation/tgstation/pull/63204
- https://github.com/tgstation/tgstation/pull/67066

### My changes
Creates a get_name component that gets the name of any mob (with 5 tile range restriction and only mobs), as /tg/ has ID getters but that's outside the scope of this PR right now. (500+ more lines)

## Why It's Good For The Game

> circuits are currently broken and useless, and this fixes that

- USBs previously did not work at all.
- We get pretty spiffy features, especially the BCIs.
- The TGUI for circuits gets an uplift, and programmable variables are a plus.

### In addition:
- Adds a `Get Name` component for mobs, this seems to be missing on /tg/ as well, and to me it just makes sense since if you put an entity into a string, it'll just be named `entity`.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog
:cl: Mothblocks, Watermelon914, read-0nly, TheSmallBlue, Gurkenglas, SmArtKar, Y0SH1M4S73R, dragomagol, ported by jupyterkat, Dejaku51, and mystery3525
refactor: Brings our wiremod (circuits) more up-to-date (August 2021ish)
add: Added more components, such as the NTnet transmitters/recievers, pressure and temperature sensors, math components, and getter/setter components
add: Brain Computer Interface component system and HUD components for them
fix: Various fixes, chiefly USB components not working, and retroactively apply fixes to the new content being brought in
/:cl:

### CODE REVIEW

File: code/modules/wiremod/shell/brain_computer_interface.dm [hunk 1]
```diff
@@ -186,8 +182,8 @@
 	charge_action = new(src)
 	bci.actions += list(charge_action)
 
-	RegisterSignal(shell, COMSIG_CARBON_GAIN_ORGAN, PROC_REF(on_organ_implanted))
-	RegisterSignal(shell, COMSIG_CARBON_LOSE_ORGAN, PROC_REF(on_organ_removed))
+	RegisterSignal(shell, COMSIG_ORGAN_IMPLANTED, PROC_REF(on_organ_implanted))
+	RegisterSignal(shell, COMSIG_ORGAN_REMOVED, PROC_REF(on_organ_removed))
```
Comment (mystery3525) [inline code/modules/wiremod/shell/brain_computer_interface.dm]: This fixes an issue in local testing where BCIs just flat out do not let you use the "Mob say" functionality (this part was not actually ported correctly from /tg/)

**Comment by Crossedfall** [comment]
You need to properly credit all of the original authors. Both in the CL and by adding them as co-authors on the changes they made. https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors

Rebase to edit the commit messages.

**Review by itsmeow — APPROVED** [review]
It's good enough that we can fix shit later if anything comes up

**Comment by itsmeow** [comment]
Wait I never merged this? what the hell

**Comment by mystery3525** [comment]
vine boom

**Pull request merged**